src/promptflow/promptflow/azure/operations/_flow_operations.py

```diff
@@ -166,59 +130,83 @@ def create_or_update(self, flow: Union[str, Path], display_name=None, type=None,
         """Validate the parameters for flow creation operation."""
+        # validate the source folder
+        logger.info("Validating flow source.")
+        if not Path(source, DAG_FILE_NAME).exists():
+            raise UserErrorException(
+                f"Flow source must be a directory with flow definition yaml '{DAG_FILE_NAME}'. "
+                f"Got {Path(source).resolve().as_posix()!r}."
+            )
+
+        # validate flow source with flow schema
+        logger.info("Validating flow schema.")
+        flow_dict = FlowOperations._validate_flow_schema(source, flow_display_name, flow_type, **kwargs)
+
+        logger.info("Validating flow creation parameters.")
         flow = load_flow(source)
         # if no flow name specified, use "flow name + timestamp"
+        flow_display_name = flow_dict.get("display_name", None)
         if not flow_display_name:
-            flow_display_name = f"{flow.display_name}-{datetime.now().strftime('%m-%d-%Y-%H-%M-%S')}"
-        elif not isinstance(flow_display_name, str):
-            raise FlowOperationError(
-                f"Flow name must be a string, got {type(flow_display_name)!r}: {flow_display_name!r}."
-            )
+            flow_display_name = f"{Path(source).name}-{datetime.now().strftime('%m-%d-%Y-%H-%M-%S')}"
 
         # if no flow type specified, use default flow type "standard"
-        supported_flow_types = FlowType.get_all_values()
+        flow_type = flow_dict.get("type", None)
         if not flow_type:
             flow_type = FlowType.STANDARD
-        elif flow_type not in supported_flow_types:
-            raise FlowOperationError(
-                f"Flow type {flow_type!r} is not supported, supported types are {supported_flow_types}"
-            )
-
-        # check description type
-        description = kwargs.get("description", None) or flow.description
 
+        # update description and tags to be the final value
+        description = flow_dict.get("description", None)
         if isinstance(description, str):
             kwargs["description"] = description
-        elif description is not None:
-            raise FlowOperationError(f"Description must be a string, got {type(description)!r}: {description!r}.")
-
-        # check if the tags type is Dict[str, str]
-        tags = kwargs.get("tags", None) or flow.tags
-        if isinstance(tags, dict) and all(
-            isinstance(key, str) and isinstance(value, str) for key, value in tags.items()
-        ):
+
+        tags = flow_dict.get("tags", None)
+        if tags:
             kwargs["tags"] = tags
-        elif tags is not None:
-            raise FlowOperationError(
-                f"Tags type must be 'Dict[str, str]', got non-dict or non-string key/value in tags: {tags}."
-            )
 
         return flow, flow_display_name, flow_type, kwargs
 
+    @staticmethod
+    def _validate_flow_schema(source, display_name=None, type=None, **kwargs):
```

Comment:
For local flow, the validation logic is implemented by inheriting `SchemaValidatableMixin`, not sure if should align here.

Reply:
I tried that first, turns out it won't raise error for warnings(e.g. unknown field) with `_validate()` of `SchemaValidatableMixin`, I guess the validation demands is different so I chose to use another method instead of changing the behavior of `SchemaValidatableMixin`.